ozone/core/api/views.py

```
@@ -461,11 +462,25 @@ def put(self, request, *args, **kwargs):
         info = Submission.objects.get(pk=self.kwargs['submission_pk']).info
         reporting_channel_name = request.data.get('reporting_channel')
         if reporting_channel_name:
-            reporting_channel_id = ReportingChannel.objects.get(
-                name=reporting_channel_name
-            ).pk
-            request.data['reporting_channel'] = reporting_channel_id
-        serializer = UpdateSubmissionInfoSerializer(info, data=request.data)
+            try:
```

Do we really need to perform this check here? Why not let it fail in the model?

---

Having thought about it more, I still don't think there's any need for this check here. The added value of the error message is almost non-existent, as the UI should, in most cases, send a proper channel here. This is not something that should normally reach the user except for some weird corner cases - not worth the extra code.